colour/recovery/tests/test_jakob2019.py

```diff
@@ -0,0 +1,188 @@
+    """
+    Defines :func:`colour.recovery.jakob2019.RGB_to_sd_Jakob2019`
+    definition unit tests methods.
+    """
+
+    def test_roundtrip_colorchecker(self):
+        """
+        Tests :func:`colour.recovery.jakob2019.RGB_to_sd_Jakob2019` definition
+        round-trip errors using a color checker.
+        """
+
+        for name, sd in COLOURCHECKER_SDS['ColorChecker N Ohta'].items():
+            # The colours aren't too saturated and the tests should pass with
+            # or without feedback.
+            for use_feedback in [False, True]:
+                XYZ = sd_to_XYZ(sd, illuminant=D65) / 100
+
+                RGB = XYZ_to_RGB(
+                    XYZ,
+                    D65_xy,
+                    colourspace.whitepoint,
+                    colourspace.XYZ_to_RGB_matrix,
+                )
+
+                recovered_sd, error = RGB_to_sd_Jakob2019(
+                    RGB,
+                    colourspace,
+                    return_error=True,
+                    use_feedback=use_feedback
+                )
+
+                if error > ACCEPTABLE_DELTA_E:
+                    self.fail('Delta E for \'{0}\' with use_feedback={1}'
+                              ' is {2}'.format(name, use_feedback, error))
+
+
+class TestErrorFunction(unittest.TestCase):
+    """
+    Defines :func:`colour.recovery.jakob2019.error_function`
+    definition unit tests methods.
+    """
+
+    EXAMPLE_LAB = np.array([72., -20., 61.])
+
+    def test_compare_intermediates(self):
+        """
+        Compares intermediate results of
+        :func:`colour.recovery.jakob2019.error_function` with
+        :func:`colour.sd_to_XYZ`, :func:`colour.XYZ_to_Lab` and checks if the
+        error is computed correctly by comparing it with
+        :func:`colour.difference.delta_E_CIE1976`.
+        """
+        # Quoted names refer to colours from ColorChecker N Ohta (using D65).
+        coefficient_list = [
+            np.array([0, 0, 0]),  # 50% gray
+            np.array([0, 0, -1e+9]),  # Pure black
+            np.array([0, 0, +1e+9]),  # Pure white
+            np.array([1e+9, -1e+9, 2.1e+8]),  # A pathological example
+            np.array([2.2667394, -7.6313081, 1.03185]),  # 'blue'
+            np.array([-31.377077, 26.810094, -6.1139927]),  # 'green'
+            np.array([25.064246, -16.072039, 0.10431365]),  # 'red'
+            np.array([-19.325667, 22.242319, -5.8144924]),  # 'yellow'
+            np.array([21.909902, -17.227963, 2.142351]),  # 'magenta'
+            np.array([-15.864009, 8.6735071, -1.4012552]),  # 'cyan'
+        ]
+
+        # error_function will not align these for us.
+        shape = DEFAULT_SPECTRAL_SHAPE_JAKOB_2019
+        aligned_cmfs = cmfs.copy().align(shape)
+        illuminant = D65.copy().align(shape)
+        illuminant_XYZ = sd_to_XYZ(D65)
+        illuminant_XYZ /= illuminant_XYZ[1]
+
+        for coefficients in coefficient_list:
+            error, derror, R, XYZ, Lab = error_function(
```

Comment:
Codacy found an issue: [Unused variable 'derror'](https://app.codacy.com/manual/colour-science/colour/pullRequest?prid=5753724)

Reply:
This is actually not me writing that but Codacy.